src/main/java/seedu/address/model/wedding/ContactMap.java

```diff
@@ -0,0 +1,108 @@
+package seedu.address.model.wedding;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import seedu.address.commons.util.ToStringBuilder;
+import seedu.address.model.person.Person;
+
+/**
+ * Represents a Contact Map of a Wedding.
+ * Guarantees: details are present and not null, field values are validated.
+ */
+public class ContactMap {
+    private HashMap<Role, Person> map;
```

Comment:
Would it be better to use Map<Role, Person> map, it may be more flexible that way (can change implementation if need be) - Since map is interface removes us from sticking to a specific implementation.